src/util/color/predefinedcolorsset.h

```diff
@@ -107,9 +109,28 @@ class PredefinedColorsSet final {
                 return color;
             }
         }
+        qWarning() << "No color matches id" << iId;
         return noColor;
     };
 
+    PredefinedColorPointer predefinedColorFromRgbColor(mixxx::RgbColor color) const {
+        for (PredefinedColorPointer pColor : allColors) {
+            if (mixxx::RgbColor(pColor->m_defaultRgba.rgb()) == color) {
+                return pColor;
+                break;
+            }
+        }
+        qWarning() << "No color matches RgbColor" << color;
+        return noColor;
+    };
+
+    PredefinedColorPointer predefinedColorFromRgbColor(mixxx::RgbColor::optional_t color) const {
+        if (!color) {
```

Comment:
Please handle `if (color)` and return noColor only as a fallback like in the non-optional case.